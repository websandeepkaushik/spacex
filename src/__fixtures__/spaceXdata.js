const spaceXdata = [
    {
        'flight_number': 1,
        'launch_year': 2006,
        'links': {
            'mission_patch_small': 'https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png',
        },
        'mission_id': [],
        'mission_name': 'FalconSat',
        'launch_success': false,
    }
];
export { spaceXdata as default };