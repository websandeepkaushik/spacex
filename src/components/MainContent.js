import React from 'react';
import { Row, Col, Image } from 'react-bootstrap';
import styled from 'styled-components';
import { useSelector } from 'react-redux';
import { getSpaceXdata } from '../state/SpaceXdata';

/**
 * This the MainContent component
 */
const MainContent = () => {
  const spaceXdata = useSelector(getSpaceXdata);

  const missionIds = (missionIds) => {
    return (missionIds && missionIds.length > 0) && 
            (
            <>
                <b>Mission Ids:</b>
                <ul>
                    {
                        missionIds.map(id => <li key={`mid-${id}`}>{id}</li>)
                    }
                </ul>
            </>
        )
  }
  return (
    <MainContainer>
        <Row>
            {
                spaceXdata?.length > 0 &&
                spaceXdata.map(item => {
                    return (
                        <Col sm={6} md={4} key={`flight-${item.flight_number}`}>
                            <ItemCell>
                                <ItemImg>
                                    <Image src={item.links.mission_patch_small} />
                                </ItemImg>
                                <H3>{`${item.mission_name} #${item.flight_number}`}</H3>
                                {missionIds(item.mission_id)}
                                <b>Launch Year: </b> {item.launch_year}<br />
                                <b>Successful Launch: </b> {item.launch_success}<br />
                                <b>Successful Landing: </b> {'data not in api'}<br />
                            </ItemCell>
                        </Col>
                    )
                })
            }
            
        </Row>
    </MainContainer>
  );
};

export default MainContent;

export const MainContainer = styled.div`
    width: 80%;
    padding-left: 30px;
    @media (max-width: 1024px) {
        .col-sm-6{
            flex: 0 0 50%;
            max-width: 50%;
        }
    }
    @media (max-width: 700px) {
        width: 100%;
        padding-left: 0;
        .col-sm-6{
            flex: 0 0 100%;
            max-width: 100%;
        }
    }
`;

export const ItemCell = styled.div`
    background: #FFF;
    padding: 10px;
    height: 100%;
    border-bottom: 20px solid #CCC;
    img{
        height: 150px;
    }

`;

export const ItemImg = styled.div`
    background: #CCC;
    text-align: center;
    margin-bottom: 10px;
    height: 150px;
`;

export const H3 = styled.h3`
    font-size: 13px;
    font-weight: bold;
    color: #24386e;
`;

export const H4 = styled.h3`
    font-size: 13px;
    font-weight: bold;
`;