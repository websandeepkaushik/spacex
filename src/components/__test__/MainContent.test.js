import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';
import MainContent from '../MainContent';
import * as redux from 'react-redux';
import store from '../../store';
import spaceXdata from '../../__fixtures__/spaceXdata';

describe('<MainContent />', () => {
  it('Renders the main view', () => {
    const {container} = render(<redux.Provider store={store}><MainContent /></redux.Provider>);
    expect(container).toMatchSnapshot();
  });

  it('Selectors test by mock store', () => {
    const sagaMiddleware = createSagaMiddleware();
    const mockStore = configureStore([sagaMiddleware]);
    const storeMock = mockStore({
        spaceXdata
    });
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((selector) => selector({spaceXReducer: storeMock.getState()}));
    const {container} = render(<redux.Provider store={store}><MainContent /></redux.Provider>);
    expect(container).toMatchSnapshot();
  });
});
