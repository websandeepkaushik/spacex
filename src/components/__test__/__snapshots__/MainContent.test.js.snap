// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContent /> Renders the main view 1`] = `
<div>
  <div
    class="sc-bdnylx dCOlis"
  >
    <div
      class="row"
    />
  </div>
</div>
`;

exports[`<MainContent /> Renders the main view 2`] = `
<div>
  <div
    class="sc-bdnylx dCOlis"
  >
    <div
      class="row"
    >
      <div
        class="col-md-4 col-sm-6"
      >
        <div
          class="sc-gtssRu fuoigp"
        >
          <div
            class="sc-dlnjPT jxzbmt"
          >
            <img
              class=""
              src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
            />
          </div>
          <h3
            class="sc-hKFyIo YcUxI"
          >
            FalconSat #1
          </h3>
          <b>
            Launch Year: 
          </b>
           
          2006
          <br />
          <b>
            Successful Launch: 
          </b>
           
          <br />
          <b>
            Successful Landing: 
          </b>
           
          data not in api
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<MainContent /> Selectors test by mock store 1`] = `
<div>
  <div
    class="sc-bdnylx dCOlis"
  >
    <div
      class="row"
    >
      <div
        class="col-md-4 col-sm-6"
      >
        <div
          class="sc-gtssRu fuoigp"
        >
          <div
            class="sc-dlnjPT jxzbmt"
          >
            <img
              class=""
              src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
            />
          </div>
          <h3
            class="sc-hKFyIo YcUxI"
          >
            FalconSat #1
          </h3>
          <b>
            Launch Year: 
          </b>
           
          2006
          <br />
          <b>
            Successful Launch: 
          </b>
           
          <br />
          <b>
            Successful Landing: 
          </b>
           
          data not in api
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<MainContent /> Selectors test by mock store with error message 1`] = `
<div>
  <div
    class="sc-bdnylx dCOlis"
  >
    <div
      class="row"
    >
      <div
        class="col-md-4 col-sm-6"
      >
        <div
          class="sc-gtssRu fuoigp"
        >
          <div
            class="sc-dlnjPT jxzbmt"
          >
            <img
              class=""
              src="https://images2.imgbox.com/3c/0e/T8iJcSN3_o.png"
            />
          </div>
          <h3
            class="sc-hKFyIo YcUxI"
          >
            FalconSat #1
          </h3>
          <b>
            Launch Year: 
          </b>
           
          2006
          <br />
          <b>
            Successful Launch: 
          </b>
           
          <br />
          <b>
            Successful Landing: 
          </b>
           
          data not in api
          <br />
        </div>
      </div>
    </div>
  </div>
</div>
`;
